src/components/settings/PushNotifications/logic.ts

```diff
@@ -0,0 +1,121 @@
+import { keccak256, toUtf8Bytes } from 'ethers/lib/utils'
+import { getToken, getMessaging } from 'firebase/messaging'
+import { DeviceType } from '@safe-global/safe-gateway-typescript-sdk'
+import type { RegisterNotificationsRequest } from '@safe-global/safe-gateway-typescript-sdk'
+import type { Web3Provider } from '@ethersproject/providers'
+
+import { FIREBASE_VAPID_KEY, initializeFirebase } from '@/services/firebase/app'
+import { trackEvent } from '@/services/analytics'
+import { PUSH_NOTIFICATION_EVENTS } from '@/services/analytics/events/push-notifications'
+import packageJson from '../../../../package.json'
+
+type WithRequired<T, K extends keyof T> = T & { [P in K]-?: T[P] }
+
+// We store UUID locally to track device registration
+export type NotificationRegistration = WithRequired<RegisterNotificationsRequest, 'uuid'>
+
+export const requestNotificationPermission = async (): Promise<boolean> => {
+  if (Notification.permission === 'granted') {
+    return true
+  }
+
+  let permission: NotificationPermission | undefined
+
+  try {
+    permission = await Notification.requestPermission()
+  } catch (e) {
+    console.error('Error requesting notification permission', e)
```

Comment:
Any reason not to use the logError function here? Is this module running in the SW?

Reply:
Addressed in bbe73f4277ad22d090b588b66f45f57e3d06d560.